Bloodhunter pacman

jurentie opened the pull request:
* Added intelligentMove to PacMan along with several helper functions for it. In intelligentMove, resolve #38 :
     * there is first a check to see if pacman is too close to a ghost, if it is it returns 'run' which calls another function runFromGhost which only takes actions away from the direction of the ghost, if possible. 
     * The only issue I run into is handling the teleports, I'm not sure how to ensure pacman can avoid the ghost when it comes through the teleport. If possible maybe someone could look into this and give ideas?
     *  So far, if no issues arise with teleport pacman seems to win almost every time.

* Also copied over code from @StaticNomad branch to keep playGame consistent.

StaticNomad commented:
What are your thoughts with getDots() in GameBoard? I saw you removed it but I thought it was needed

jurentie commented:
@StaticNomad I renamed getDots() -> setDots() to initialize the dots and then created getDots() to return the array of dot objects to use in PacMan.py.
I also added a line in GameBoard.py to remove the dots from the array as needed.

StaticNomad commented:
Okay, I'm going to merge this and starting working on any conflicts that arise in Pull Request #37
